text/1055-vanilla-prettier-setup-in-blueprints.md

```diff
@@ -0,0 +1,93 @@
+---
+stage: accepted
+start-date: 2024-12-03T00:00:00.000Z
+release-date: 
+release-versions:
+teams:
+  - cli
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/1055
+project-link:
+suite: 
+---
+
+# Vanilla Prettier Setup in Blueprints
+
+## Summary
+
+This RFC proposes to migrate to a vanilla Prettier setup in the blueprints, instead of running Prettier via ESLint and Stylelint.
+
+This RFC also proposes to not include any custom values for specific Prettier options, and instead truly rely on Prettier's defaults for formatting.
+
+## Motivation
+
+1. Because we run Prettier via ESLint and Stylelint, we only run the files these linters support through Prettier - Using a vanilla Prettier setup, would format all files Prettier supports, ensuring even more consistency in projects
+2. Less dependencies in the blueprints - `eslint-plugin-prettier` and `stylelint-prettier` would not be needed anymore
+3. The Prettier team recommends running Prettier directly, and not via linters:
+  - Running Prettier directly is faster than running Prettier via ESLint and Stylelint
+  - ESLint and Stylelint show red squiggly lines in editors (when using the corresponding extensions), while the idea behind Prettier is to make developers forget about formatting
+4. Not including any custom values for specific Prettier options makes the blueprints less biased, and introduces more consistency - I.e. double quotes for _all_ files, instead of single quotes for `.cjs`, `.cts`, `.gjs`, `.gts`, `.js`, `.mjs`, `.mts` and `.ts` files, and double quotes for the rest - To me, this makes more sense, especially for new projects
+
+`3.` is mostly taken from [Integrating with Linters > Notes](https://prettier.io/docs/en/integrating-with-linters.html#notes)
+
+## Detailed Design
+
+We would add the following scripts to the `package.json` file in the `app` blueprint:
+
+```diff
++ "lint:format": "prettier . --cache --check",
```

Comment:
I think these should be just "format" and "format:fix".

We don't want to run these in parallel with the other lint tasks, because prettier can conflict with eslint and template-lint rules that _kinda_ format -- resulting in a flip-flopping of formatting depending on which eslint or template-lint.

What I do in my projects is this:
```js
"lint": "concurrently 'npm:lint:*' && npm run format",
"lint"fix": "concurrently 'npm:lint:*:fix' && npm run format:fix",
```

prettier has to have the final say when other tools can adjust positions of characters.

Reply:
I see your point about running the fixes in a particular order, but using `lint:format` instead of `format` is what makes this a safe drop-in upgrade that doesn't cause people to lose CI coverage.

Since the default CI setup runs `lint:*`, no change to CI is needed to make it check `lint:format`.

Reply:
default `test` command (which i don't agree with, but I don't have the energy to fight it) is
```bash
    "test": "concurrently \"npm:lint\" \"npm:test:*\" --names \"lint,test:\"",
```

so running `npm:lint` would include format with the above suggestion/modification, and keep order correct.

Reply:
I think @NullVoxPopuli is correct, but maybe the script should be `format` and `format:check` then or something? `pnpm format` sounds like it's already going to do the actual formatting of all the files.

Reply:
We could do `lint:format` and `format` to address both points. It has the trade-off not following the established `lint:*:fix` convention.

As an alternative we could also change the implementation of `lint` to ensure a specific order, can't we?

Reply:
`lint:format` and `format` sounds good to me too.

Reply:
Updated to use `lint:format` and `format`. I also added that we will update the `lint:fix` script to always run `format` last.